tests/struct_loader.rs

```diff
@@ -215,3 +216,55 @@ fn test_struct_loader_load_orders() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_struct_loader_into_iterator() -> Result<()> {
+    let base_dir = get_test_base_dir();
+    let empty_dict = Dict::<String>::new();
+
+    let mut loader = StructLoader::<Item>::new("items.yml", &base_dir);
+
+    {
+        // when items are not loaded
+        let unloaded_item_iter = loader.into_iter();
+        assert_eq!(unloaded_item_iter.count(), 0);
+    }
+
+    loader.load(&empty_dict)?;
+
+    {
+        // when items are loaded
+        let item_iter = loader.into_iter();
+        for (name, item) in item_iter {
+            match name.as_str() {
+                "Melon" => {
+                    assert_eq!(item.name, "melon");
+                    assert_eq!(item.price, 500.0);
+                }
+                "Orange" => {
+                    assert_eq!(item.name, "orange");
+                    assert_eq!(item.price, 200.0);
+                }
+                "Apple" => {
+                    assert_eq!(item.name, "apple");
+                    assert_eq!(item.price, 100.0);
+                }
+                "Carrot" => {
+                    assert_eq!(item.name, "carrot");
+                    assert_eq!(item.price, 150.0);
+                }
+                _ => panic!("unexpected item"),
+            }
+        }
+
+        let item_iter = loader.into_iter();
+        let item_names: Vec<&str> = item_iter.map(|(_, item)| item.name.as_str()).collect();
+        assert_eq_unordered!(item_names, vec!["melon", "orange", "apple", "carrot"]);
```

Comment:
I understand you're testing below 1, 2 separately.

1. whether each record's attributes matches each other
2. if the records contains all the elements needed (in unordered way)

These two parts are basically testing one thing, so it would be nice if we can test the both each one by one so that test demonstrates explicit expectations. (you could sort them by ids, for instance, to examine the records in order)